File: 0771-jewels-and-stones/0771-jewels-and-stones.py
from collections import defaultdict

class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        stoneCount = defaultdict(int)
        for i in stones:
            stoneCount[i] += 1
        ans = 0
        for i in jewels:
            ans += stoneCount[i]
        return ans
    

File: 0771-jewels-and-stones/0771-jewels-and-stones.java
class Solution {

    public int numJewelsInStones(String jewels, String stones) {
        String[] split = stones.split("");

        int answer = 0;
        for (int i = 0; i < split.length; i++) {
            if (jewels.contains(split[i])) {
                answer++;
            }
        }

        return answer;
    }
}